### ISSUE TITLE
After key does not trigger loading of dependent plugin if using only cond key

### ISSUE THREAD

**Issue by akinsho** [wbthomason/packer.nvim#272]
Apologies for the maybe confusing title, also since I think you're planning on a general change to how this works this might be moot but I thought pointing out this specific issue was worth doing anyway. 

Essentially if a plugin needs to be loaded after another, it isn't currently correctly loaded if the plugin it is to be loaded after is loaded after a `cond`. I'd imagin that the condition is run pretty much immediately i.e. during packer compiled's execution, but the after never seems to kick in if the condition is true and that plugin is then loaded

real world eg.

```lua
-- This doesn't work i.e. flutter tools is not loaded
use {"neovim/nvim-lspconfig", cond = function() return vim.env.ENV_VAR ~= nil end}
use  {"akinsho/flutter-tools", after = "nvim-lspconfig", config = function() print('loaded!!') end}
-- this does work
use {"neovim/nvim-lspconfig"}
-- Or
use {"neovim/nvim-lspconfig", event = "VimEnter"}
use  {"akinsho/flutter-tools", after = "nvim-lspconfig", config = function() print('loaded!!') end}
```

**Comment by akinsho** [comment]
On digging in a bit this now makes much more sense and I just didn't understand. The `after` key is *only* relevant when a plugin is being loaded via an autocommand otherwise it does nothing. So it can't be used to control the order of execution for start packages for example 🤔 . Not sure if vim just loads all of these arbitrarily in a random order once installed or if packer can intervene.

Might just be my perspective but I think being able to guarantee ordering for plugins that aren't necessarily lazy loaded is pretty important. I think even if it means maybe making these plugins optional and then loading them during the packer compiled's execution but then doing it sequentially based on the after key etc. Not sure if this deviates from any of what you were thinking, or you have a better idea in mind @wbthomason

**Comment by wbthomason** [comment]
Wow, this slipped my radar. No, `after` should determine sequencing always, not just for autocommand loading. It technically cannot control `start` package order, but only because it implies `opt` - it can absolutely control eagerly-loaded package ordering. This is definitely a bug in the compiler, which might take some redesign to do.

> I think even if it means maybe making these plugins optional and then loading them during the packer compiled's execution but then doing it sequentially based on the after key etc

Good suggestion! This is in fact what we already do :smile: 

The trick is that we currently load `cond` plugins with a direct `packadd`; this should really use the load module function we use for e.g. autocommand lazy-loads to ensure that the chain of sequence-loads gets updated.

**Comment by gwerbin** [comment]
> It technically cannot control `start` package order, but only because it implies `opt` - it can absolutely control eagerly-loaded package ordering.

Would it be totally ridiculous to group `pack/packer/start/*` directories into subdirectories to enforce ordering?
```
pack/packer/start/
  01/
    pluginA/
    pluginB/
  02/
    pluginC/
  99/
    pluginD/
```

**Comment by akinsho** [comment]
I believe this was closed by #390 see https://github.com/wbthomason/packer.nvim/pull/390#issuecomment-921955871
